### ISSUE TITLE
Issue with Creating a Date Search Field in CKAN Facet with Custom Start and End Dates

### ISSUE THREAD

**Issue by sedama29** [ckan/ckan#8306]
## CKAN version
CKAN 2.10

## Describe the bug
I am trying to create a date search field in the facet with custom start and end dates.

### Steps to reproduce
Attempt to create a date search field in CKAN facet.
Specify custom start and end dates for the search.

### Expected behavior
I expected to successfully implement a date search field in the CKAN facet where users can specify custom start and end dates to filter datasets.

**Comment by wardi** [comment]
@JVickery-TBS is our date range hack on the publishing queue search something that could be shared as an example?

**Comment by JVickery-TBS** [comment]
@wardi Yeah, I think there is some improvement to be made in the Core code to make this stuff more frameworkable.

As we had to add code into `ckan.lib.search.query` and `ckan.logic.action.get`:
- https://github.com/open-data/ckan/blob/canada-v2.10/ckan/lib/search/query.py#L34
- https://github.com/open-data/ckan/blob/canada-v2.10/ckan/lib/search/query.py#L505
- https://github.com/open-data/ckan/blob/canada-v2.10/ckan/logic/action/get.py#L1975
- https://github.com/open-data/ckan/blob/canada-v2.10/ckan/logic/action/get.py#L1980

So I think we would need to be able to have an implement method for `VALID_SOLR_PARAMETERS` and further to define facet ranges in the `package_search` action method. Unless if you can already do the latter with `dataset_facets` from `IFacets` interface class??

If we can already do the `package_search` part with `IFacets`, then I think we can just add the `'facet.range.end', 'facet.range', 'facet.range.gap', 'facet.range.start'` to the `VALID_SOLR_PARAMETERS`? And possibly at the same time, add a `valid_solr_parameters` method to `IFacets` interface class?

**Comment by Nisha1293** [comment]
Hi @JVickery-TBS  @wardi 
I am working on this issue.

**Comment by Nisha1293** [comment]
@sedama29 
Could you please share more detail to reproduce the issue or any screenshot ?

**Comment by sedama29** [comment]
I want to add two fields, Start Date and End Date, to the dataset extras. I also want to include these fields in the search filter on the left side of the Datasets page, allowing users to filter datasets by Start Date and End Date.
